lib/sailor.js

```diff
@@ -210,23 +210,21 @@ class Sailor {
     }
 
     scheduleShutdown() {
-        return co(function* doScheduleShutdown() {
-            if (this.shutdownCallback) {
-                log.debug('scheduleShutdown – shutdown is already scheduled, do nothing');
-                return new Promise(resolve => this.shutdownCallback = resolve);
-            }
-
-            yield this.amqpConnection.listenQueueCancel();
-            if (this.messagesCount === 0) {
-                // there is no unfinished processMessage invocation, let's just resolve scheduleShutdown now
-                log.debug('scheduleShutdown – about to shutdown immediately');
-                return;
-            }
-            // at least one processMessage invocation is not finished yet
-            // let's return a Promise, which will be resolved by processMessageAndMaybeShutdownCallback
-            log.debug('scheduleShutdown – shutdown is scheduled');
+        if (this.shutdownCallback) {
```

Comment:
no new logic => just kill co
?

Reply:
this.amqpConnection.listenQueueCancel() => this.amqpConnection.stopConsume();